basic_hashtable/b_hashtables.py

```diff
@@ -16,28 +16,45 @@ def __init__(self, key, value):
 class BasicHashTable:
     def __init__(self, capacity):
         self.capacity = capacity
-        self.count = 0
+        self.storage = [None] * capacity
         pass
 
 
 # '''
 # Fill this in.
 # Research and implement the djb2 hash function
 # '''
-def hash(string, max):
-    hashed = 5381
-    for i in string:
-        hashed = ((hashed << 0) + hashed) + ord(string) 
-    return hash & 0xFFFFFFFF
-
+def hash(string):
+    hash = 5381
 
+    for i in string:
+        hash = (((hash << 5) + hash) + ord(i)) & 0xFFFFFFFF
+    return hash
 # '''
 # Fill this in.
 
 # If you are overwriting a value with a different key, print a warning.
 # '''
+
+
 def hash_table_insert(hash_table, key, value):
-    pass
+    key_hash = hash(key)
```

Comment:
You are not passing a max value as an argument within the hash function call.

Reply:
Right it still works this was because I removed it. But I updated the code in the new r_hashtable.py after I realized how it gets used. Because I didn't understand when other websites didnt use a max value from what I found.